### DISCUSSION TITLE
Different highlight groups for titles of search and cmdline

### DISCUSSION THREAD

**Discussion by JacquesCherbuin** [folke/noice.nvim#739] [Q&A] [answered]
The search and cmdline views have different highlight groups for the border and icons, but they share the same highlight group for the title "NoiceCmdlinePopupTitle". How can I set different colors for the search title and cmdline title? 
![Screen Shot 2024-02-24 at 10 32 20 am](https://github.com/folke/noice.nvim/assets/63825183/9d9f133f-a027-4d2e-91c0-d069f0e25fab)
![Screen Shot 2024-02-24 at 10 32 10 am](https://github.com/folke/noice.nvim/assets/63825183/0913db71-7996-4f1d-a170-244b6917c027)

**Comment by Sam-programs** [comment, marked as answer]
You can specify a view per format.
```lua
format = {
  cmdline = { ... , view = "cmdline_normal" },
}
```

You can create custom views from the examples in [noice.config.views](https://github.com/folke/noice.nvim/blob/main/lua/noice/config/views.lua#L30), or copy them and modify `FloatTitle`.

Here is an exmaple for making the cmdline's title highlight `Visual`:
```lua
local cmdline_normal = vim.deepcopy(require('noice.config.views').defaults.cmdline_popup)
cmdline_normal.win_options.winhighlight.FloatTitle = "Visual"
require("noice").setup({
    views = {
        cmdline_normal = cmdline_normal,
    },
    cmdline = {
        -- Default view for other formats
        view = "cmdline_popup",
        format = {
            cmdline = { pattern = "^:", icon = "", lang = "vim", view = "cmdline_normal" },
        },
    },
})
```

**Reply by JacquesCherbuin** [reply]
Thanks, I understand now. As a follow up. is it possible to activate cmdline with other events instead of a pattern? The issue is that I'm trying to create a view for git commit. My current config is:
```lua 
--Keybinding
map("n", "<space>gc", ':Git commit -m ""<Left>')

-- noice.setup()
-- ...
git_commit = {
	pattern = "^:Git commit %-m",
	icon = "",
	view = "cmdline_normal",
	opts = {
		size = {
			height = "auto",
			width = "50%",
		},
		position = {
			row = "50%",
			col = "50%",
		},
		border = {
			text = { top = "Commit" },
			style = "single",
		},
	},
},
-- ...
```
Which yields:
![Screen Shot 2024-02-26 at 09 15 14 am](https://github.com/folke/noice.nvim/assets/63825183/cb8cc2c9-2323-4d31-915d-3b677d59677f)

Is there anyway to use vim.fn.input() to trigger noice cmdline so that I can hide the double quotes in the cmdline prompt?

**Reply by Sam-programs** [reply]
> Is there anyway to use vim.fn.input() to trigger noice cmdline so that I can hide the double quotes in the cmdline prompt?

Yea but the styling is limited to one format for all `vim.fn.input` calls.
https://github.com/folke/noice.nvim/blob/bf67d70bd7265d075191e7812d8eb42b9791f737/lua/noice/ui/cmdline.lua#L72-L76
There is no pattern checking it just picks it.
However you could make a custom command that doesn't need quotes that invokes `Git commit` with quotes.
```lua
vim.api.nvim_create_user_command("GitCommit", function(opts)
  vim.cmd("Git commit -m \"" .. opts.args .. "\"")
end, {
    -- 	-nargs=1    Exactly one argument is required, it includes spaces
    nargs = 1
})
```

**Reply by Sam-programs** [reply]
Oh i just remembered there is a parser for git commits, it stops highlighting when the commit is too long so you could use that instead of resizing the window.
```lua
gitcommit = {
    pattern = "^:GitCommit ",
    lang = "gitcommit",
    ...
},
```

**Reply by JacquesCherbuin** [reply]
Awesome, thanks a lot.
